backends/utils_test.py

```diff
@@ -25,147 +20,141 @@
-            "expires_in": 31535999  # 1 year - 1 second
-        }
-        social_user = self.update_social_extra_data(extra_data)
-        utils.refresh_user_token(social_user)
-        assert not mock_refresh.called
-
-    @patch('backends.edxorg.EdxOrgOAuth2.refresh_token', return_value=social_extra_data, autospec=True)
-    def test_refresh_400_error_server(self, mock_refresh):
-        """Test to check what happens when the OAUTH server returns 400 code"""
-        def raise_http_error(*args, **kwargs):  # pylint: disable=unused-argument
-            """Mock function to raise an exception"""
-            error = HTTPError()
-            error.response = MagicMock()
-            error.response.status_code = 400
-            raise error
-
-        mock_refresh.side_effect = raise_http_error
-        social_user = self.user.social_auth.get(provider=EdxOrgOAuth2.name)
-        with self.assertRaises(utils.InvalidCredentialStored):
-            utils._send_refresh_request(social_user)
-
-    @patch('backends.edxorg.EdxOrgOAuth2.refresh_token', return_value=social_extra_data, autospec=True)
-    def test_refresh_401_error_server(self, mock_refresh):
-        """Test to check what happens when the OAUTH server returns 401 code"""
-        def raise_http_error(*args, **kwargs):  # pylint: disable=unused-argument
-            """Mock function to raise an exception"""
-            error = HTTPError()
-            error.response = MagicMock()
-            error.response.status_code = 401
-            raise error
-
-        mock_refresh.side_effect = raise_http_error
-        social_user = self.user.social_auth.get(provider=EdxOrgOAuth2.name)
-        with self.assertRaises(utils.InvalidCredentialStored):
-            utils._send_refresh_request(social_user)
-
-    @patch('backends.edxorg.EdxOrgOAuth2.refresh_token', return_value=social_extra_data, autospec=True)
-    def test_refresh_500_error_server(self, mock_refresh):
-        """Test to check what happens when the OAUTH server returns 500 code"""
-        def raise_http_error(*args, **kwargs):  # pylint: disable=unused-argument
-            """Mock function to raise an exception"""
-            error = HTTPError()
-            error.response = MagicMock()
-            error.response.status_code = 500
-            raise error
-
-        mock_refresh.side_effect = raise_http_error
-        social_user = self.user.social_auth.get(provider=EdxOrgOAuth2.name)
-        with self.assertRaises(HTTPError):
-            utils._send_refresh_request(social_user)
-
-
-class SocialTests(TestCase):
+pytestmark = pytest.mark.django_db
+
+
+@pytest.fixture
+def user():
+    """Create a user with social auth"""
+    # create a user
+    user = UserFactory.create()
+    user.social_auth.create(
+        provider='not_edx',
+    )
+    user.social_auth.create(
+        provider=EdxOrgOAuth2.name,
+        uid="{}_edx".format(user.username),
+        extra_data=social_extra_data
+    )
+    yield user
```

Comment:
you have a similar user fixture in `backends/pipeline_api_test.py`, would it make sense for that to live in a test util library or something instead?

Reply:
Sounds good, I just combined the two and moved it to `fixtures/common.py`